direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/RetryStrategy.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.direct.blob;
+
+import com.google.common.base.Preconditions;
+import cz.o2.proxima.annotations.Internal;
+import cz.o2.proxima.functional.Factory;
+import cz.o2.proxima.util.ExceptionUtils;
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Internal
+public class RetryStrategy implements Serializable {
```

Comment:
can we replace this with failsafe?
https://github.com/jhalterman/failsafe

Reply:
Actually no, the code is straightforward and it is better not to pull in another dependency if we don't have to.